Added B matrix functionality

nwhel opened the pull request:
Added support for primitive internal coordinates, Wilson B matrices, and generating trajectories using these.

nwhel committed:
Added B matrix functionality

Added support for primitive internal coordinates, Wilson B matrices, and generating trajectories using these.

nwhel committed:
Added mypy typing to B matrix functions, updated docstrings

nwhel committed:
Updated docs to reflect new docstrings

nwhel committed:
Added typing-extensions to setup.py

nwhel committed:
Moved TypeAlias from typing to typing-extensions

typing does not have TypeAlias before Python 3.12, moved for backwards compatability.

nwhel committed:
Attempted fix for | in Python<=3.9

added "from __future__ import annotations"

nwhel committed:
Changed type | type to Union[type, type]

nwhel committed:
Added () to function call

nwhel committed:
Changed typing to be more readable

Added Vector, Matrix, Tensor3D types to differentiate between NDArrays

nwhel committed:
Fixed Python 3.9 support

Changed type union from | to Union[]

nwhel committed:
Fixed numba tying issue, added cache=True

nwhel committed:
Changed IO and fixed angle delta minimization

Now go through each element of delta_c and minimize the paths between angles.

mcocdawc commented:
- [ ] break into functions
- [ ] pure internal movement (?)   ([get_inertia](https://chemcoord.readthedocs.io/en/v2.1.2/src_Cartesian/src_Cartesian/chemcoord.Cartesian.get_inertia.html) and [get_barycenter](https://chemcoord.readthedocs.io/en/v2.1.2/src_Cartesian/src_Cartesian/chemcoord.Cartesian.get_barycenter.html#chemcoord.Cartesian.get_barycenter) might be useful)
- [ ] decorator wrapper for mypy and numba.njit, passing on `__annotation__`
- [ ]

mcocdawc committed:
removed `get_B_traj` but keep the Wilson B

was a great idea, too bad it did not work. :-(

nwhel committed:
Added tests

nwhel committed:
Removed unnecessary numba typing

nwhel committed:
improved cartesian position indexing

for get_Wilson_B and x_to_c, now pos_arr is an n by 3 array where n is the number of atoms containing the positions of each atom with normal indexing.

nwhel committed:
Clarified variable names and changed to numpy array indexing

nwhel committed:
Replaced lists with arrays, updated variable names for readability

nwhel committed:
Test for function compartmentalization.

nwhel committed:
Fixed method arguments for test

nwhel committed:
Added type hint

nwhel committed:
Compartmentalized B matrix calculation

nwhel committed:
Fixed type hinting

nwhel reviewed (commented):
Mislabeled this, meant to say fixed method arguments.

nwhel committed:
Fixed array constructor

nwhel committed:
trying to fix array constructor

mcocdawc committed:
Merge branch 'master' of github.com:mcocdawc/chemcoord into B_matrix_implementation

mcocdawc committed:
added a new_primitive_coords method

mcocdawc committed:
added known to fail test

mcocdawc commented:
already done with all the latest merges from private
